lib/Lift.cpp

```diff
@@ -510,8 +510,10 @@ namespace {
 static llvm::APInt ReadValueFromMemory(const uint64_t addr, const uint64_t size,
                                        const remill::Arch *arch,
                                        const Program &program) {
-  llvm::APInt result(size, 0);
-  for (auto i = 0u; i < (size / 8); ++i) {
+
+  // create an instance of precision integer of size*8 bits
+  llvm::APInt result(size * 8, 0);
+  for (auto i = 0u; i < size; ++i) {
     auto byte_val = program.FindByte(addr + i).Value();
     if (remill::IsError(byte_val)) {
       LOG(ERROR) << "Unable to read value of byte at " << std::hex << addr + i
```

Comment:
Reset the error log to `std::dec` after `(addr + i)`.

Reply:
Otherwise all subsequent logs will log out line numbers in hex, which is unhelpful.